buildkite/scripts/rosetta-integration-tests.sh

```diff
@@ -222,100 +176,56 @@ until [ $daemon_status == "Synced" ]; do
-      gets updated, this build will most likely fail and we will then
-      need to update the zkapp-cli version used here. This is
-      unfortunate, but necessary. Please delete this warning once it's
-      done." > /dev/stderr
-
-deploy_txs=()
-for zkapp_path in ${ZKAPP_PATH}/*/; do
-  zkapp_path=${zkapp_path%/}
-  zkapp=$(basename $zkapp_path)
-  echo "Deploying ${zkapp}..."
-
-  zkapp_account_pk=$(cat ${MINA_KEYS_PATH}/zkapp-${zkapp}-account.key.pub)
-  zkapp_account_privkey=$(mina-dev advanced dump-keypair --privkey-path "${MINA_KEYS_PATH}/zkapp-${zkapp}-account.key" | sed -ne "s/Private key: //p")
-
-  mkdir -p ${zkapp_path}/keys
-  echo -e "{\n    \"privateKey\": \"${zkapp_account_privkey}\",\n    \"publicKey\": \"${zkapp_account_pk}\"\n}" >"${zkapp_path}/keys/sandbox.json"
-
-  cat <<EOF >"${zkapp_path}/config.json"
-{
-  "version": 1,
-  "networks": {
-    "sandbox": {
-      "url": "http://127.0.0.1:${MINA_GRAPHQL_PORT}/graphql",
-      "keyPath": "keys/sandbox.json",
-      "feepayerKeyPath": "/root/.cache/zkapp-cli/keys/sandbox.json",
-      "feepayerAlias": "sandbox",
-      "fee": "1"
-    }
-  }
-}
-EOF
-  cd "$zkapp_path"
-  npm ci
-  npm run build
-  txn=$(zk deploy sandbox -y | sed -ne "s/https:\/\/berkeley.minaexplorer.com\/transaction\///p")
-  deploy_txs+=txn
-  cd -
-  echo "Done."
-done
-
-# TODO: wait until all zkApps deploy txns are included in a block
+send_value_transfer_txns &
 
-next_block_time=$(mina-dev client status --json | jq '.next_block_production.timing[1].time' | tr -d '"')
-curr_time=$(date +%s%N | cut -b1-13)
-sleep_time=$((($next_block_time - $curr_time) / 1000))
-echo "Sleeping for ${sleep_time}s until next block is created..."
-sleep ${sleep_time}
-
-# Start calling zkApp methods
-echo "==================== INTERACTING WITH ZKAPPS ======================"
-RECEIVER_PK=$BLOCK_PRODUCER_PK
-for zkapp_path in ${ZKAPP_PATH}/*/; do
-  zkapp_path=${zkapp_path%/}
-  zkapp=$(basename $zkapp_path)
-  echo "Interacting with ${zkapp}..."
-
-  cd "$zkapp_path"
-  ./interact.sh sandbox
-  cd -
-  echo "Done."
-done
+# Start sending zkapp transactions
+ZKAPP_FEE_PAYER_NONCE=1
+ZKAPP_SENDER_NONCE=1
+ZKAPP_STATE=0
+send_zkapp_transactions() {
+  while true; do
+    ZKAPP_TXN_QUERY=$(zkapp_test_transaction transfer-funds-one-receiver --fee-payer-key ${ZKAPP_FEE_PAYER_KEY} --nonce ${ZKAPP_FEE_PAYER_NONCE} --sender-key ${ZKAPP_SENDER_KEY} --sender-nonce ${ZKAPP_SENDER_NONCE} --receiver-amount 1 --fee 5 --receiver ${ZKAPP_ACCOUNT_PUB_KEY} | sed 1,5d)
+    send_zkapp_txn "${ZKAPP_TXN_QUERY}"
+    let ZKAPP_FEE_PAYER_NONCE++
+    let ZKAPP_SENDER_NONCE++
+
+    ZKAPP_TXN_QUERY=$(zkapp_test_transaction update-state --fee-payer-key ${ZKAPP_FEE_PAYER_KEY} --nonce ${ZKAPP_FEE_PAYER_NONCE} --zkapp-account-key ${ZKAPP_SENDER_KEY} --zkapp-state ${ZKAPP_STATE} --fee 5 | sed 1,5d)
+    send_zkapp_txn "${ZKAPP_TXN_QUERY}"
+    let ZKAPP_FEE_PAYER_NONCE++
+    let ZKAPP_STATE++
+  done
+}
+send_zkapp_transactions &
```

Comment:
Shouldn't we wait until these transactions are indeed sent to the network (or better still, until they're included in a block) before we move on to calling `rosetta-cli`? In the current shape the test moves so fast that sometimes `rosetta-cli` starts before even block # 2 is produced…